### ISSUE TITLE
Advance communication platform

### ISSUE THREAD

**Issue by a-teammate** [inexor-game/code#352]
We are currently on IRC and sometimes on Mumble.

IRC has no persistent log and is pretty awkward to get going, its the standard platform for oss devs though (but we are on gamesurge, which is basically dead).

There are alternatives:
* ~~slack~~ (closed source)
* ~~discord~~ (closed source) #343
* [Lets chat](https://sdelements.github.io/lets-chat/)
* [Rocket.Chat](https://rocket.chat/)
* [mattermost](https://about.mattermost.com/)

**Comment by Croydon** [comment]
1) I think we should switch to freenode (like serveral people already requested in the past), TLS, bigger network
2) Why do we need logs and what is it good for?
3) If we want logs then we could still use something like https://botbot.me (we definitely shouldn't host any more services on our own, since we already have a huge struggle to keep the current ones up)
4) People can get always free bouncers, e.g. here is a list: http://wiki.znc.in/Providers and they can use web IRC (linked in our readme)
5) I was playing with the idea to set up a **bi-directional syncing** between our IRC channel and a (new) Telegram group. Telegram has at least open APIs and official open source clients (the server code is closed source). And it is certainly more END-USER-friendly than IRC. I see our IRC channel mainly as a channel for developers, testers, bug reporters etc. but this may just improve our accessible so it can't be that bad. (And it's not like we have so many interested people that we have to separate). Telegram has also (relative) many user. I never heard of these three platforms you just linked. And since Telegram would be an addition to IRC nobody would be forced to use it.

I still don't really see the point why we should switch to a completely new setup. As you said it's still a de-facto standard, despite all the seemingly shiny alternatives.

Your issue title says team communication, but I guess it's the channel for everyone then. Splitting it even more up in different places (team some alternative, others IRC or so) would be insane.

**Comment by a-teammate** [comment]
These are all workarounds imo.
So much work for so little output.
We shouldn't spend time on developing IRC bots, we shouldn't spend time on doing stuff which is obsolete with better alternatives.

I agree: OSS people are used to IRC on freenode. Which is a big plus for IRC.

However many more professionals are used to feature-rich stuff like slack, simply bc its awesome.
Some OSS people are professionals as well, I don't think they're overburdened by registering to a "new" service (maybe they use it already).

Look at https://github.com/RocketChat/Rocket.Chat#features
and try https://demo.rocket.chat: it basically makes mumble obsolete as well (+ Teamviewer/anything for screensharing which is a BIG plus if you do pair-programming).

Notice: I don't think this will split up the community from the team communication, basically the opposite:
Currently there are different, not well communicated channels on IRC (can you find all?).
Channel management will be much easier with rocket.chat

Logs using a bouncer is not comparable with permanent logs: you don't have logs from the past if you joined the team later on.

**Comment by Croydon** [comment]
> We shouldn't spend time on developing IRC bots, we shouldn't spend time on doing stuff which is obsolete with better alternatives.

We don't. We use existing solutions.

> Currently there are different, not well communicated channels on IRC (can you find all?).

What are you talking about? AFAIK we only have one IRC channel, if this is not the case then the communication itself fails, not the platform. 

> Logs using a bouncer is not comparable with permanent logs: you don't have logs from the past if you joined the team later on.

My point about bouncers has nothing to do with logs. At some point you communicated that the missing of bouncers is a negative point for using IRC. So I pointed out that everyone can use a bouncer without a problem.

You still didn't answer why we need logs. And as I said there are good ways to get logs going if we want to have some.

**Comment by a-teammate** [comment]
I think there are two choices:

Either: Have a place where oss-people are used to

Or: Require people to (O)Auth to a new application for the benefit of not having to worry about communication anymore.

**Comment by aschaeffer** [comment]
I think we should stay on IRC for **developer** communication. We just have one channel and this won't change in near future. Don't forget, we have GitHub issues, which are our "channels". Also, IRC is multi-platform. We already have Mumble for voice chat - also done. Therefore I don't see much benefit from rocket chat - but administration overhead.

* :-1: for create a whole new chat infrastructure (now)
* :+1: for permanent IRC logs
* :+1: for move to freenode
* :+1: for using issues, wiki and git

End user communication is a completely different thing.

**Comment by Croydon** [comment]
I can fully recommend riot.im as an addition to our IRC channel.

With Riot people can directly connect to our IRC channel, but have some advanced features like
 * file uploading
 * voice/video calling
 * notifiactions when mentioned (mobile, web, email)

File uploading is compatible with IRC, it's getting posted as a link to the file.
Voice/Video calling is not possible with IRC anyway, so IRC people won't miss that. In other words IRC+Riot works perfectly together.

Riot looks more user friendly than many IRC web chats or bouncer user interfaces, so it may help some less technical people in getting started as well.

Personally I use now the Riot Android app for my Nooby_mobile account instead an IRC app with an rather ugly UI.

The mentioning feature is really nice to reach out to people faster and from the own perspective getting new information faster. This may also help with people who can't be often around: they won't miss imporant stuff that easily anymore.

People can give it a try without registering. Here is a direct link: https://riot.im/app/#/room/#inexor:matrix.org

I trust the project enough to say that we don't need to host it on our own (also we shouldn't, we don't have the time and resources).

Riot seems to be a fast developing project so even I have seen some very minor flaws I'm overall very positive about it. It could replace Mumble, however, the project itself sees group voice calling right now as unstable (but who knows what the means before someone tried it, and it may become stable soon).

My only condition is that IRC stays our central point. So third party integrations should (if possible) happen directly to IRC please.

**Comment by Fohlen** [comment]
Just to keep this updated, there's now a Rocket.Chat instance at https://chat.inexor.org !

**Comment by Croydon** [comment]
When you all agree I would request logging at https://botbot.me

Then we have some kind off official log. E.g. Riot or IRC bouncers can provide more personal logs.

**Comment by a-teammate** [comment]
ok
my thoughts:
* Rocket.Chats UI is intuative
* Rocket.Chat is pretty slow.
* Rocket.Chat is stupid to set up
   * even the github integration just works randomly (they recommend hubot instead)
   * IRC bridge also only works sparsely 
-----
* Riot seems much more responding
* Riot's room people list seems to grow unlimited if connect to IRC
* Riot's UI is a bit better than raw IRC
  * better formatting of the messages
* but it still sucks
  * most stuff is hidden counter-intuatively
     * dont believe me? try to open and close the peoples list
  * separation between different messages is pretty hard to see
* Riot wont replace pastebin (RocketChat would)
  * since its quoting syntax doesn't work with newlines ` ``` `
  * or does it? see next point
* Riot is totally undocumented
  * you wont even find it on google if not searching with additional keywords

and last but not least the most problematic point with Riot:
* Riot has no historical logs
  * everything gets documented from the point on you joined the room
    * same as a bouncer in IRC


In conclusion Riot seems (atm) like a bouncer/client combi to access IRC.
Its far from mature though, but Rocket.Chat isn't either.
Rocket.Chat is really really slow (mostly loading takes ours, the mobile app does not even cache its connection for me)
Its still lacking a lot of features.
Ive no idea, i would still choose Rocket.Chat for serious dev communication, since it provides stuff Riot doesn't.
But i hate how slow it is. Maybe we could check out mattermost?
Or we wait for Riot to become more than an experimental [Matrix] client, but use it already

**Comment by Croydon** [comment]
> dont believe me? try to open and close the peoples list

IMO not a problem

> you wont even find it on google if not searching with additional keywords

They had a rename not long ago so that may still affect things.

> Riot has no historical logs
    everything gets documented from the point on you joined the room
        same as a bouncer in IRC

I don't really understand the need for that. Decision making should happen on GitHub. IRC should be mainly stuff important for the moment, to keep things going, but stuff which is likely to be totally uninteresting in a few days. Even more when you weren't even involved in the project at this time. But anyway; I created the room differently and now it has permanent, from the beginning logs. Just join here: https://riot.im/app/#/room/#inexor:matrix.org

**Comment by Croydon** [comment]
>Riot wont replace pastebin (RocketChat would)
    since its quoting syntax doesn't work with newlines ```
    or does it? see next point

It works with tree triple backticks like you wrote. On web inserting new lines is possible with Shift+Enter or of course copy+paste. On mobile the respective keyboard should have such an option.

> Or we wait for Riot to become more than an experimental [Matrix] client, but use it already

Let's go for that.

**Comment by Fohlen** [comment]
As @Croydon already mentioned using

- [Telegram](https://telegram.org/)
- [Signal](https://play.google.com/store/apps/details?id=org.thoughtcrime.securesms&hl=de)

would be as well an option for us. Both of the messengers bring the benefits of:

- beeing open-source
- beeing hosted
- having a broad API that let's us use GitHub and IRC bots
- are slightly popular among developers, I myself already use Telegram

On the con side is:

- they are beeing hosted, means less freedom on our side
- Telegram[ has been claimed to be hacked](https://www.reddit.com/r/crypto/comments/5na901/has_russia_cracked_telegram_messenger/), though imho we don't really post sensitive stuff there anyways

Anyways I'd like it as a more permanent solution that we can use on mobil to connect more easily.

**Comment by Fohlen** [comment]
I have added a Telegram channel likewise for testing, join it at https://t.me/inexor

**Comment by a-teammate** [comment]
Did you check out mattermost yet @Fohlen ?
https://gitlab.mattermost.com/community/channels/town-square
it seems pretty equal to RocketChat but without the huge performance penalty

btw do we use https://sandstorm.io to get RocketChat or the usual way?
Too bad there is no mattermost plugin for sandstorm yet id say

**Comment by Croydon** [comment]
While we are still testing other platforms, can we agree on officially recommending Riot.im as an addition to IRC?

Btw, did someone test the voice chat feature yet?

**Comment by a-teammate** [comment]
i think IRC should really only be recommended for people who already have a bouncer on freenode.
Anyone else should get Riot. otherwise there is no improvement in connectivity (people join ask sth, leave 5 min later.. )*

But I'd still favor mattermost over all alternatives (including matrix) for all developers:

matrix is nice as an extended IRC. But its not much more. if you look at how projects like [mattermost itself](https://pre-release.mattermost.com/core/channels/town-square) organize their development AND community talk and additional stuff over mattermost.. thats a different level.


_*(the mixed room is distracting: one can not really use the handful of matrix/riot features since you don't know how it will look in irc. + the IRC bridge is often just useless, since its never showing disconnects)
IRC should be the last choice._

**Comment by Croydon** [comment]
> But I'd still favor mattermost over all alternatives (including matrix) for all developers:

Will check that out soon.

>the mixed room is distracting: one can not really use the handful of matrix/riot features since you don't know how it will look in irc. 

I wrote above something about that https://github.com/inexor-game/code/issues/352#issuecomment-268112781
Anything I didn't mention and you worried about?

>the IRC bridge is often just useless, since its never showing disconnect

This is an highly active work-in-progress issue for them. User list syncing is already enabled for smaller IRC networks but Freenode still needs some work. Big channels like #ubuntu had issues with Matrix join/disconnecting spam in the past and they need to work on that before they activate it. But the problem will go away in a foreseeable future.

**Comment by Croydon** [comment]
> As @Croydon already mentioned using
    Telegram
    Signal

I didn't mention Signal, nor do I think it would be fitting.

>Both of the messengers bring the benefits of:
    beeing open-source
    beeing hosted
    having a broad API that let's us use GitHub and IRC bots
    are slightly popular among developers, I myself already use Telegram

Only the Telegram clients are open source, not the server. 
Being hosted is definitely an advantage, but so are many other alternatives.
The GitHub bot did spam too much. It posted all kind of information instead of just a summary. Too distracting from user messages.
Popularity is indeed a big plus.


I like Telegram in general, but I don't see how we can could integrate it well right now.
There are same libraries to sync IRC with Telegram, but that would mean to host something again by ourselves and these libraries  I saw so far doesn't look very stable.

For Riot/Matrix integrations are a top priority and they are working on Telegram integration as well. They are considered not stable enough to be configurable on the Riot UI for now.

When Riot/Matrix have a Telegram integration Telegram would be nice addition I guess, before this happens I'm against making Telegram an official channel.
Doing so right now has only the potentional to split up the communication and people are missing stuff on all platforms.

**Comment by Fohlen** [comment]
I think that usually the most used platform is the one that in the end will determine what is "official".
_Hiding_ the channel in tickets however, so that someone must manually ask me to join the channel ( @movabo did so), is really bad I think.
This as well should be more in focus on the new website.

**Comment by a-teammate** [comment]
settled on riot/matrix.
close
